### PR TITLE
basic comments

### PR DESCRIPTION
**Pull request by bml2238**
files i did not document:
ArtifactHandler.java
artifact.service.ts
image-upload.component.ts
user-page.component.ts (already had comments, but not official ones)

### CODE REVIEW

File: frontend/crd/src/app/portfolio/portfolio.component.ts [hunk 1]
```diff
@@ -55,6 +58,9 @@ export class PortfolioComponent implements OnInit{
     });
   }
 
+  /**
+   * Opens the file select modal
```
Comment (jts7382) [inline frontend/crd/src/app/portfolio/portfolio.component.ts:62]:
```suggestion
   * Opens the Submission Modal
```

File: frontend/crd/src/app/portfolio/portfolio.component.ts [hunk 2]
```diff
@@ -46,6 +46,9 @@ export class PortfolioComponent implements OnInit{
     });
   }
 
+  /**
+   * The student is able to upload their resume directly from the portfolio
```
Comment (jts7382) [inline frontend/crd/src/app/portfolio/portfolio.component.ts:50]:
```suggestion
   * Grabs the Student's artifacts to be displayed.
   * Currently just grabs the resume
```

File: frontend/crd/src/app/util/task.service.ts [hunk 3]
```diff
@@ -17,8 +17,13 @@ export class TaskService {
   ) {
   }
 
+  /**
+   * Gets all the tasks and caches the response
+   * 
+   * If the cache has data in it, it returns the value of the cache, otherwise
+   * it makes a request to the backend.
```
Comment (jts7382) [inline frontend/crd/src/app/util/task.service.ts:24]:
```suggestion
   * it makes a request to the backend.
   * @param forceRefresh forces the cache to update by sending the request again
```

**Review by jts7382 — CHANGES_REQUESTED** [review]

File: frontend/crd/src/app/admin/task-main-page/task-main-page.component.ts [hunk 4]
```diff
@@ -56,7 +60,7 @@ export class TaskMainPageComponent implements OnDestroy {
     const modalDialog = this.matDialog.open(TaskEditModalComponent, dialogConfig);
 
     modalDialog.afterClosed().subscribe(result => {
-      //TODO: successful save popup?
```
[the remark this answers is not in the document]
Comment (bml2238) [reply frontend/crd/src/app/admin/task-main-page/task-main-page.component.ts:59]: there are save popups in the modals themselves but we could add one on the main page later

**Review by JohnLDavidson — APPROVED** [review]

**Review by jts7382 — APPROVED** [review]

**Pull request merged**